lexer/lexer.hpp

```diff
@@ -1,12 +1,45 @@
+#pragma once
+
+#include <functional>
+#include <cctype>
 #include "trie.hpp"
 #include "token.hpp"
-#include "trie.hpp"
-#pragma once
 
-namespace lexer
-{
-    std::vector<Token> read(std::string);
-    common::Trie<TokenType> getTrie();
-    std::vector<Token> splitLine(std::string);
-    void freeResult();
-} // namespace lexer
\ No newline at end of file
+namespace lexer {
+
+
+extern common::Trie<TokenType> g_keywordTrie;
+extern common::Trie<TokenType> g_operatorTrie;
+
+class Lexer {
+public:
+    Lexer(std::string src) : m_buf(src) {}
+
+    Token Next();
+
+private:
+    std::string m_buf;
+    size_t m_pos = 0, m_lastAfterNewLine = 0;
+    size_t m_lineNum = 1;
+
+    static bool isspace(char ch) {
+        return ch != '\n' && std::isspace(ch);
```

Comment:
Consider using [`isblank`](https://en.cppreference.com/w/cpp/string/byte/isblank) instead. As I understand, it's the same without the new-line/carriage-return characters

Reply:
Yeah, but carriage-returns _are_ the problem. I chose to treat them as space everywhere since they preceded the newline characters. This way, on Windows and on any other OS, the files will be scanned correctly.

Reply:
Actually, when reading the file in C/C++, you do not see `\r` at all. Windows removes it automatically when reading and writes it to the file automatically when writing `\n`. It only exists (transparently) for backwards compatibility.
So unless we need to support typewriters, there is no need to worry about CRLF)

Reply:
> anybody could read the file in binary mode

Who and how? this module is only to be used in out compiler :D